mkosi/__init__.py

```diff
@@ -278,6 +278,22 @@ def configure_os_release(context: Context) -> None:
             if context.config.image_version and not image_version_written:
                 new.write(f'IMAGE_VERSION="{context.config.image_version}"\n')
 
+        # On dpkg distributions we cannot simply overwrite the file as it is owned by a package.
+        # We need to set up a diversion first, so that it is not overwritten by package updates.
+        if context.config.distribution.is_apt_distribution():
+            run([
+                "dpkg-divert",
+                "--quiet",
+                "--root",
+                context.root,
+                "--local",
+                "--add",
+                "--rename",
+                "--divert",
+                f"/{candidate}.mkosi",
+                f"/{candidate}"
+            ], stderr=subprocess.DEVNULL)
```

Comment:
We've never pretended that running a package manager on mkosi produced images was going to do the right thing in the first place. If you put `IMAGE_ID` and `IMAGE_VERSION` in /etc/os-release, I think it's safe to assume we're doing a golden image that won't be updated via package manager anymore.

Reply:
Is dpkg going to overwrite /etc/os-release? If not can't we rework the os-release stuff to also replace /etc/os-release so we don't have to do the dpkg-divert stuff?

Reply:
Yes, as it's a symlink in a package, so it would be overwritten. And also we could be building a usr-only image, but as above we can't know here, as it's only specified in repart.conf, so we have to change it in /usr/

Reply:
I'm suggesting we change both /etc/os-release (replace symlink with our own file) and /usr/lib/os-release. Then on update /usr/lib/os-release gets overwritten but /etc/os-release won't be overwritten. If dpkg overwrites /etc/os-release on updates that's a bug to fix in debian

Reply:
But what's the point of doing that? You need a divert in place to avoid it being replaced anyway in either case

Reply:
Wow, I didn't want to believe this, but just gave it a try and `/etc/os-release` is clobbered when `base-files` is installed. That behaviour is a bug that should be fixed, since that's config that is admin territory.

Reply:
> But what's the point of doing that? You need a divert in place to avoid it being replaced anyway in either case

Policy is that /etc/os-release should always be preferred over /usr/lib/os-release, so there's no need for the divert if we also replace /etc/os-release since it won't matter what the contents of /usr/lib/os-release are.

Yes this doesn't handle the case where you're doing upgrades on a /usr only image but I don't think we need to support that use case. Either you do /usr only golden image or you do regular image with /etc included and this approach covers both those use cases.

Reply:
> > But what's the point of doing that? You need a divert in place to avoid it being replaced anyway in either case
> 
> Policy is that /etc/os-release should always be preferred over /usr/lib/os-release, so there's no need for the divert if we also replace /etc/os-release since it won't matter what the contents of /usr/lib/os-release are.
> 
> Yes this doesn't handle the case where you're doing upgrades on a /usr only image but I don't think we need to support that use case. Either you do /usr only golden image or you do regular image with /etc included and this approach covers both those use cases.

ok that's probably good enough